roles/replace_node/tasks/volume.yml

```diff
@@ -0,0 +1,113 @@
+---
+# Set up the volume management
+- name: Fetch the directory and volume details
+  block:
+    - name: Get the list of volumes on the machine
+      shell: ls "{{ glusterd_libdir }}/vols"
+      register: dir_list
+
+    - set_fact:
+        volumes: "{{ dir_list.stdout.split() }}"
+
+    # Find the list of bricks on the machine
+    - name: Get the list of bricks corresponding to volume
+      shell: >
+        gluster vol info {{ item }} | grep "Brick.*{{ item }}:" |
+        awk -F: '{ print $3 }'
+      with_items: "{{ volumes }}"
+      register: brick_list
+
+    - set_fact:
+        volume_bricks: {}
+
+    - name: Get the list of bricks into a variable
+      set_fact:
+        volume_bricks: "{{ volume_bricks|combine({item.item: item.stdout}) }}"
+      loop: "{{ brick_list.results }}"
+  delegate_to: "{{ gluster_maintenance_cluster_node }}"
+
+- name: Create the brick directory if not already present
+  file:
+    state: directory
+    path: "{{ item.value }}"
+    mode: 0755
+  loop: "{{ lookup('dict', volume_bricks) }}"
+  delegate_to: "{{ gluster_maintenance_new_node }}"
+
+- name: Set the volume attributes
```

Comment:
Set the volume attributes using reset-brick?

Reply:
Yes, reset-brick command sets the volume extended attributes. No? 

Since "Reset the GlusterFS brick" seemed a bit confusing.

Reply:
Ack! Done.